### ISSUE TITLE
`allow_attributes` Lint does not trigger on `#[allow(warnings)]`

### ISSUE THREAD

**Issue by kamulos** [rust-lang/rust-clippy#13425]
### Summary

This is probably a bit weird, but I found this in some code: `#[allow(warnings)]` annotated on a function to mute all warnings in it. Now I am using the `clippy::allow_attributes` Lint and expected to be forced to write `expect`. But nothing like this happened.

### Lint Name

allow_attributes

### Reproducer

I tried this code:

```rust
#[allow(warnings)]
fn foo(input: &str) {
    let _first_value = input.split(',').nth(0).unwrap();
}
```

I expected to see this happen:

clippy telling me to write `expect` instead of `allow`

Instead, this happened:

the warning was muted so the `allow(warnings)` seems to be something valid, but clippy did not instruct me to use `expect`

**Comment by kamulos** [comment]
Hmm I also raised https://github.com/rust-lang/rust/issues/130609 which calls into question if `expect(warnings)` makes sense at all. So this could be the correct behavior by clippy

**Comment by samueltardieu** [comment]
Running the nightly Clippy on the following code:

```rust
#![deny(clippy::allow_attributes)]

#[allow(warnings)]
fn foo(input: &str) {
    let _first_value = input.split(',').nth(0).unwrap();
}

fn main() {
}
```

gives me

```
error: #[allow] attribute found
 --> src/main.rs:3:3
  |
3 | #[allow(warnings)]
  |   ^^^^^ help: replace it with: `expect`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#allow_attributes
note: the lint level is defined here
 --> src/main.rs:1:9
  |
1 | #![deny(clippy::allow_attributes)]
  |         ^^^^^^^^^^^^^^^^^^^^^^^^
```

Isn't that what you are expecting?
